README.md

```diff
@@ -1,3 +1,164 @@
+```
+
+Or perhaps we want to know how `y` will change with respect to changes of
+rotation angle:
+```julia
+∂y_∂θ = transform_deriv_params(rot, x)
+```
+
+### The interface
+
+Transformations are derived from `AbstractTransformation{OutType, InType}`.
+`InType` is a (possibly abstract or union) type describing which inputs
+`transform()` will accept with the given transformation, and `OutType` describes
+the (possible range of) outputs given. These parameters allow for a level of
+safety, so that transformations are only applied to appropriate data types and
+to catch early-on any problems with composing or chaining transformations.
+
+As an example, we have `Translation{T} <: AbstractTransformation{FixedVector, FixedVector}`.
+A translation will expect data in the `FixedVector` format and will always return
+the same base type as given (of course, type promotion may occur for the element type itself).
+
+Transformations can be reversed using `inv(trans)`. They can be chained
+together using the `∘` operator (`trans1 ∘ trans2`) or `compose` function (`compose(trans1, trans2)`).
+In this case, `trans2` is applied first to the data, before `trans1`.
+Composition may be intelligent, for instance by precomputing a new `Translation`
+by summing the elements of two existing `Translation`s, and yet other
+transformations may compose to the `IdentityTransformation`. But by default,
+composition will result in a `ComposedTransformation` object which simply
+dispatches to apply the transformations in the correct order.
+
+Finally, the matrix describing how differentials propagate through a transform
+can be calculated with the `transform_deriv(trans, x)` method. The derivatives
+of how the output depends on the transformation parameters is accessed via
+`transform_deriv_params(trans, x)`. Users currently have to overload these methods,
+as no fall-back automatic differentiation is currently included. Alternatively,
+all the built-in types and transformations are compatible with automatic differentiation
+techniques, and can be parameterized by *DualNumbers*' `DualNumber` or *ForwardDiff*'s `GradientNumber`.
+
+### Built-in transformations
+
+A small number of 2D and 3D coordinate systems and transformations are included.
+We also have `IdentityTransform{InOutType}` and `ComposedTransformation{InType,OutType}`,
+which allows us to nest together arbitrary transformations to create a
+complex yet efficient transformation chain.
+
+#### Coordinate system types
+
+The canonical Cartesian coordinate systems are any type which inherited from
+*FixedSizeArrays*' `FixedVector{2}` and `FixedVector{3}` in 2D and 3D,
+respectively. The concrete datatypes `Point` and `Vec` are provided by
+*FixedSizeArrays*, and here we tend to favor `Point` for output where it is not
+specified. (Of course, users may define their own transformations and types, as
+they wish).
```

Comment:
Again, there seems to be a useful distinction between coordinate systems and coordinates - should this be 

**Points in** the canonical Cartesian coordinate systems are any type which inherit from ...